packages/next/src/cli/next-dev.ts

```diff
@@ -392,12 +419,175 @@ If you cannot make the changes above, but still want to try out\nNext.js v13 wit
+          if (
+            cleanChunk.match(
+              /(ENOENT|Module build failed|Module not found|Cannot find module)/
+            )
+          ) {
+            if (startDir === dir) {
+              try {
+                // check if start directory is still valid
+                const result = findPagesDir(
+                  startDir,
+                  !!config.experimental?.appDir
+                )
+                shouldFilter = !Boolean(result.pagesDir || result.appDir)
+              } catch (_) {
+                shouldFilter = true
+              }
+            }
+            if (shouldFilter || startDir !== dir) {
+              shouldFilter = true
+              return
+            }
+            process[fd].write(chunk)
+          } else {
+            process[fd].write(chunk)
+          }
+        }
+
+        for (const workerId in cluster.workers) {
+          cluster.workers[workerId]?.process.stdout?.on('data', (chunk) => {
+            filterForkErrors(chunk, 'stdout')
+          })
+          cluster.workers[workerId]?.process.stderr?.on('data', (chunk) => {
+            filterForkErrors(chunk, 'stderr')
+          })
+        }
+      }
+
+      const handleClusterExit = () => {
+        const callback = async (worker: cluster.Worker) => {
+          // ignore if we killed the worker
+          if ((worker as any).killed) return
+
+          // TODO: we should track how many restarts are
+          // occurring and how long im-between them
+          if (worker.exitedAfterDisconnect) {
+            setupFork()
+          } else if (!sessionStopHandled) {
+            await handleSessionStop()
+            process.exit(1)
+          }
+        }
+        cluster.addListener('exit', callback)
+        return () => cluster.removeListener('exit', callback)
+      }
+      let clusterExitUnsub = handleClusterExit()
+      cluster.settings.stdio = ['ipc', 'pipe', 'pipe']
+
+      setupFork()
+      config = await loadConfig(PHASE_DEVELOPMENT_SERVER, dir)
+
+      const handleProjectDirRename = (newDir: string) => {
+        clusterExitUnsub()
+
+        for (const workerId in cluster.workers) {
+          try {
+            // @ts-expect-error custom field
+            cluster.workers[workerId].killed = true
+            cluster.workers[workerId]!.process.kill('SIGKILL')
+          } catch (_) {}
+        }
+        process.chdir(newDir)
+        // @ts-expect-error type is incorrect
+        cluster.settings.cwd = newDir
+        cluster.settings.exec = cluster.settings.exec?.replace(dir, newDir)
+        setupFork({
+          ...Object.keys(process.env).reduce((newEnv, key) => {
+            newEnv[key] = process.env[key]?.replace(dir, newDir)
+            return newEnv
+          }, {} as typeof process.env),
+          NEXT_PRIVATE_DEV_DIR: newDir,
+        })
+        clusterExitUnsub = handleClusterExit()
+      }
+      const parentDir = path.join('/', dir, '..')
```

Comment:
Are you sure the `/` prefix is correct? I would think this could cause problems on windows